Commit for code-review

nikaqim opened the pull request:
This repo is created for code-review for walkthrough component to empty main

Commit:
Commit for code-review

web-fullstack-developer commented on src/app/app.component.html:
```diff
@@ -0,0 +1,7 @@
+  <router-outlet />
+</main>
+
+<link href="https://fonts.googleapis.com/css2?family=Calistoga&display=swap" rel="stylesheet" />
```
The font files should be linked in index.html, not here.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/tree/6eba859d2a52fa24c6c049752d12d75f3627604a

web-fullstack-developer commented on src/app/app.component.ts:
```diff
@@ -0,0 +1,16 @@
+  styleUrl: './app.component.scss'
+})
+export class AppComponent {
+  // title = 'reusetuto';
```
Removing unused and commented-out code helps keep the codebase clean and maintainable.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/tree/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer commented on src/app/app.module.ts:
```diff
@@ -0,0 +1,36 @@
+import { StartScreenModule } from './components/start-screen/start-screen.module';
+import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
+
+const config:SocketIoConfig = {
```
For better readability and maintainability, configuration details should be defined in a separate file and imported wherever needed.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/06017d2df044eebea8f4cecbeb955ba5260ad553

web-fullstack-developer commented on src/app/app-routing.module.ts:
```diff
@@ -0,0 +1,27 @@
+  },
+  {
+      path: 'start',
+      component: StartScreenComponent
```
For better performance, please use lazy loading modules

nikaqim replied:
completed -> https://github.com/nikaqim/FVR_RETUTO_PR/tree/9671fef3bcf19f76dc9f502691c0befaac6072e0

web-fullstack-developer commented on src/app/components/start-screen/start-screen.module.ts:
```diff
@@ -0,0 +1,13 @@
+    SharedModule
+  ]
+})
+export class StartScreenModule { }
```
It's recommended to use standalone components instead of NgModule. So, instead of defining the StartScreenModule to declare the StartScreenComponent, you can convert StartScreenComponent into a standalone component.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/706762051b92b06b2e242f64f1c0b3f6f9064666

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.html:
```diff
@@ -0,0 +1,33 @@
+        </ng-walkthrough>
+
+        <ng-template #step1Template>
+            <!-- <lib-icon-btn class="closebtn" (click)="close()">
```
remove commented code if not needed

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/1ae3d4b691431705e779c87d5d68cf73b0a1b086

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.html:
```diff
@@ -0,0 +1,33 @@
+<div #tutorWrapper class="tutorial-container">
+    <ng-container *ngFor="let step of steps">
```
it's better to use the new @ngfor syntax instead of <ng-container *ngFor="let step of steps">

nikaqim replied:
completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/34418ddc27b27a159428ae201eefa2ecced0d688

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.scss:
```diff
@@ -0,0 +1,80 @@
+    overflow-y: clip;
+    height: 80px;
+
+    ul[_ngcontent-ng-c3498083930] {
```
what is this "_ngcontent-ng-c3498083930"?

nikaqim replied:
remove rendered attribute id in css -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/784ad73967f47171be67690e0a12c11f4938b38f

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,362 @@
+@Component({
+  selector: 'app-cyrano-walkthrough',
+  templateUrl: './cyrano-walkthrough.component.html',
+  styleUrl: './cyrano-walkthrough.component.scss'
```
For better performance, please use the OnPush change detection strategy. This will help optimize rendering by ensuring that Angular only checks for changes when necessary, reducing unnecessary re-renders.

nikaqim replied:
completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/438fc188a536a8d627cac9a8a997fd3fd62678d4

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,362 @@
+    ngOnInit(): void {
+
+      // rxjs observable
+      this.subs.add(
```
For better readability and more structured code, please move the initialization code into a private method and call it inside ngOnInit(). This will help keep the component's lifecycle hook clean and maintainable.

nikaqim replied:
completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/7c18986b30b4415f3712c8410eb87cb139baabf0

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,362 @@
+      );
+    }
+
+    reset(config:CyranoTutorialConfig=this.data){
```
If the function is not used in the HTML template and is intended for internal use only, it's a good practice to add the private access modifier to the function. This ensures that the function is only accessible within the class and isn't exposed for use in the template or elsewhere.

nikaqim replied:
completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/9627abd217449ae09e3a7be49d47ee17d43db7dd

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,362 @@
+      this.construct_walk();
+    }
+
+    drawArrow(event:WalkthroughComponent, arrowId: string){
```
same here. it should be private

nikaqim replied:
completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/9627abd217449ae09e3a7be49d47ee17d43db7dd

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,362 @@
+    /**
+   * Assigning walkthrough attributes setting value
+   */
+  construct_walk(): void {
```
same here

nikaqim replied:
completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/9627abd217449ae09e3a7be49d47ee17d43db7dd

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.ts:
```diff
@@ -0,0 +1,122 @@
+    return false;
+  }
+
+  highlightAll(screenIsActive:boolean){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/222d095af8ce022241381d5ae990f95e9a26e6e9

nikaqim replied:
Adding access mod -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/244f83230821a0eec587e7154f6ae9da1415b721

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group.component.ts:
```diff
@@ -0,0 +1,126 @@
+  this.ButtonGroupReady.emit(this.buttonIds.join())
+  }
+
+  getButtonPosition(index: number, total: number) {
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/199a6f526ca58e0d738bb676d9f7870e1370174f

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group.component.ts:
```diff
@@ -0,0 +1,126 @@
+    };
+  }
+
+  assignMain(btn:Button){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/commit/199a6f526ca58e0d738bb676d9f7870e1370174f

nikaqim replied:
Adding access mod -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/5788b925722504b176fcb1b8beecab96934ccd23

web-fullstack-developer commented on src/app/components/shared/shared.module.ts:
```diff
@@ -0,0 +1,61 @@
+// import { CyranoTutorialModule } from 'cyranoTutorial';
+
+// AoT requires an exported function for factories
+export function HttpLoaderFactory(http: HttpClient) {
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/ebdf0677d1bbf740ea4be9dd830050d8092bce1d

web-fullstack-developer commented on src/app/components/start-screen/start-screen.component.ts:
```diff
@@ -0,0 +1,65 @@
+    // console.log("onInputChnge:this.steps",this.steps)
+  }
+
+  reverseMarkup(descr:string){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/cd3a07adb6990595d22d7cf00bc42d3360d29ed7

nikaqim replied:
Adding access modfier -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/01b552711684f713ef3323b4d38bb16633251385

web-fullstack-developer commented on src/app/services/btn.service.ts:
```diff
@@ -0,0 +1,66 @@
+      this.btnGroupReadySubject.next(btnIds);
+    }
+
+    onNotifyButtonReady(){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/a815972b337b38ef4ae2ba59e7834b6a14e4c4ee

nikaqim replied:
Adding access modifier -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/f9cd6638920528ec7733b1f54d0f2b8476c06135

web-fullstack-developer commented on src/app/services/btn.service.ts:
```diff
@@ -0,0 +1,66 @@
+      this.loadButtonConfig()
+    }
+  
+    getButtonConfig(): Observable<BtnGroupConfig> {
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/a815972b337b38ef4ae2ba59e7834b6a14e4c4ee

nikaqim replied:
Adding access modifier -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/f9cd6638920528ec7733b1f54d0f2b8476c06135

web-fullstack-developer commented on src/app/services/btn.service.ts:
```diff
@@ -0,0 +1,66 @@
+      return ids;
+    }
+
+    getScreenContainerId(elementId:string){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/a815972b337b38ef4ae2ba59e7834b6a14e4c4ee

nikaqim replied:
Adding access modifier -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/f9cd6638920528ec7733b1f54d0f2b8476c06135

web-fullstack-developer commented on src/app/services/i18n.servise.ts:
```diff
@@ -0,0 +1,24 @@
+        this.translateService.use(lang);
+    }
+
+    getCurrentLanguage(): string {
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/f9cd6638920528ec7733b1f54d0f2b8476c06135

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    this.walkthroughs.delete(id);
+  }
+
+  getParentPosition(child: HTMLElement, targetClass:string){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    return null;
+  }
+
+  getWalkhroughData(): Observable<CyranoTutorialConfig> {
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    }
+  }
+
+  getConfig(){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    return this.walkconfig;
+  }
+
+  implementArrMarkup(descr:WalkDescr[]): WalkDescr[]{
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    return rtnMarkups;
+  }
+
+  implementMarkUp(descr:string){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    return rtnMarkups;
+  }
+
+  reverseMarkUp(descr:string){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    this.localStorage.setData(StorageId.WalkConfig, '');
+  }
+
+  onFinishLoadWalkThru(){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    this.swiperNavSubject.next(this.getScreenById(id));
+  }
+
+  onTutoNavigation(){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,361 @@
+    this.closeTutoSubject.next(true);
+  }
+
+  onTutoClose(){
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/2457ea9b58e1f3da90750ee572bd6a54e3e91bb7

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.ts:
```diff
@@ -0,0 +1,122 @@
+@Component({
+  selector: 'app-main-screen',
+  templateUrl: './main-screen.component.html',
+  styleUrl: './main-screen.component.scss'
```
For better performance, please use the OnPush change detection strategy.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/c26b421376b225537989930474b7f8fed6ca2269

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group.component.ts:
```diff
@@ -0,0 +1,126 @@
+@Component({
+  selector: 'app-btn-group',
+  templateUrl: './btn-group.component.html',
+  styleUrl: './btn-group.component.scss'
```
For better performance, please use the OnPush change detection strategy.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/dc9b07bd1207508d1f8d099d73ed4a5a83537b1c

web-fullstack-developer commented on src/app/components/shared/language-selector/language-selector.component.ts:
```diff
@@ -0,0 +1,24 @@
+@Component({
+  selector: 'app-language-selector',
+  templateUrl: './language-selector.component.html',
+  styleUrl: './language-selector.component.scss'
```
For better performance, please use the OnPush change detection strategy.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/c8863ab016b011cb8cca4f61f72c6bcfb18a0fad

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.module.ts:
```diff
@@ -0,0 +1,16 @@
+    CyranoWalkthroughComponent
+  ]
+})
+export class CyranoWalkthroughModule { }
```
same here can be standalone component

nikaqim replied:
https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/706762051b92b06b2e242f64f1c0b3f6f9064666

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.html:
```diff
@@ -0,0 +1,56 @@
+
+                <div class="btn-container {{walkthroughActive !== '' ? 'onwalk': ''}} {{ highlightAll(isActiveScreen(panel)) ? 'highlightall' : ''}}">
+                    <!-- to be made into a component -->
+                     <div *ngFor="let btngrp of buttonGroup">
```
why need this <div> can't add the *ngFor="let btngrp of buttonGroup"  on the <app-btn-group> ?

nikaqim replied:
sorry i dont get this question. can you please describe more ?

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.html:
```diff
@@ -0,0 +1,56 @@
+        </div>
+
+        <div class="mobile screen-panel">
+            <div class="screen-container {{walkthroughActive !== '' ? 'onwalk': ''}}" [id]="'mob'">
```
can use [class.onwalk]="!!walkthroughActive" instead of {{walkthroughActive !== '' ? 'onwalk': ''}} it will be more clean

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/8853b3dcacfedc5683974cbfee968f0849c90b55

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.html:
```diff
@@ -0,0 +1,56 @@
+                    <img src="/assets/imgs/closebtn.png" />
+                 </lib-icon-btn>
+                 
+                <div class="btn-container {{walkthroughActive !== '' ? 'onwalk': ''}}" >
```
same here

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/8853b3dcacfedc5683974cbfee968f0849c90b55

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.scss:
```diff
@@ -0,0 +1,181 @@
+.page.main-page {
```
remove empty class

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/53f51b7a1b1a19d755a89fbb00a52ae07fd44ee7

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.scss:
```diff
@@ -0,0 +1,181 @@
+    position: relative;
+    display: flex;
+    justify-content: center;
+    // justify-content: space-between;
```
remove commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer commented on src/app/services/i18n.servise.ts:
```diff
@@ -0,0 +1,24 @@
+
+@Injectable({ providedIn: 'root' })
+export class I18nService {
+    availableLanguages = [
```
Instead of duplicating this list here and in MainScreenComponent, it would be better to define it in a separate file and import it both here and in MainScreenComponent

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/13033fffe52dea70ad58357fb2934c4882d98f0e

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.ts:
```diff
@@ -0,0 +1,122 @@
+  ngOnInit(): void {
+
+    // for testing realtime update using websocket
+    this.subs.add(
```
same here move the init code into private function and call it here

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/222d095af8ce022241381d5ae990f95e9a26e6e9

web-fullstack-developer commented on src/app/components/main-screen/main-screen.module.ts:
```diff
@@ -0,0 +1,20 @@
+import { ButtonsModule } from 'nextsapien-component-lib';
+
+import { MainScreenComponent } from './main-screen.component';
+// import { TutoWalkthroughModule } from '../tuto-walkthrough/tuto-walkthrough.module';
```
remove commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/commit/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group-config.model.ts:
```diff
@@ -0,0 +1,5 @@
+import { ButtonGroup } from "./btn-group.model";
+
+export interface BtnGroupConfig {
```
Rename it to IBtnGroupConfig for better naming convention.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/43a88f0e0b6c4212aaacb828dc6e75da4727b1d3

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group.component.scss:
```diff
@@ -0,0 +1,93 @@
+            align-items: center;
+            transition: all 0.3s ease-in-out;
+        }
+        // > div:nth-child(1) {
```
clean commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/commit/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group.model.ts:
```diff
@@ -0,0 +1,31 @@
+import { Button } from "../button/button.model";
+
+export interface mainAssigned {
```
Rename it to IMainAssigned

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/08d84d8bfd154feb24f8df01e6301c182289e25f

web-fullstack-developer commented on src/app/components/shared/button/button.component.html:
```diff
@@ -0,0 +1,14 @@
+<div [ngClass]="'custom-btn ' + btnSetting.size" (click)="buttonClicked(btnSetting.action)">
+    <!-- <lib-icon-btn [id]="screenId !== 'mob' ? (screenId + btnSetting.id).toLowerCase() : btnSetting.id">
```
clean commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/commit/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/4a7ba23e509f92c580d6f731e923a870d90e3251

web-fullstack-developer commented on src/app/components/shared/button/button.component.scss:
```diff
@@ -0,0 +1,101 @@
+.custom-btn {
+    // display: flex;
```
same here

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/commit/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer commented on src/app/components/shared/button/button.component.ts:
```diff
@@ -0,0 +1,75 @@
+import { TranslateService } from '@ngx-translate/core';
+import { Button } from './button.model';
+import { WalkthroughConfigService } from '../../../services/tuto.service';
+// import { CyranoTutorialService } from 'cyranoTutorial';
```
same here

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/commit/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer commented on src/app/components/shared/button/button.component.ts:
```diff
@@ -0,0 +1,75 @@
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrl: './button.component.scss'
```
Missing change detection strategy "onPush"

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/7fc489e50afc137a7cd10cfaa5c528c94cf8ad33

web-fullstack-developer commented on src/app/components/shared/button/button.component.ts:
```diff
@@ -0,0 +1,75 @@
+  ngAfterViewInit(): void {
+  }
+
+  buttonClicked(type: string) {
```
Define the return type and access modifier for this method.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/6f86532f09a46dd9170eba7e22b819a8f7fe5d54

web-fullstack-developer commented on src/app/components/start-screen/start-screen.component.ts:
```diff
@@ -0,0 +1,65 @@
+      this.steps = JSON.parse(JSON.stringify(tmp));
+  }
+
+  ngAfterViewInit(): void {
```
remove empty lifecycle method

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/b6301f147d425a0ad56242950e13b1abf81782af

web-fullstack-developer commented on src/app/components/start-screen/start-screen.component.ts:
```diff
@@ -0,0 +1,65 @@
+    return this.walkService.reverseMarkUp(descr);
+  }
+
+  ngOnDestroy(): void {
```
same here

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/b6301f147d425a0ad56242950e13b1abf81782af

web-fullstack-developer commented on src/app/services/ws.service.ts:
```diff
@@ -0,0 +1,68 @@
+        this.socket.connect();
+    }
+
+//     private connectWs() {
```
clean commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/commit/8f784ef6dff3d3f8020eedaaa1ae593f907af09d

web-fullstack-developer reviewed (changes requested):
Please go through all the files, add the missing return types and access modifiers to the methods, and clean up any commented code and empty methods. All interfaces should follow the naming convention, starting with 'I'.

Commit:
The font files should be linked in index.html, not here.

Commit:
Removing unused and commented-out code helps keep the codebase clean and maintainable.

Commit:
web-fullstack-developer 3 days ago
For better readability and maintainability, configuration details should be defined in a separate file and imported wherever needed.

Commit:
For better performance, please use lazy loading modules

Commit:
remove commented code if not needed

Commit:
it's better to use the new @ngFor syntax instead of <ng-container *ngFor="let step of steps">

Commit:
what is this "_ngcontent-ng-c3498083930"?

Commit:
For better performance, please use the OnPush change detection strategy. This will help optimize rendering by ensuring that Angular only checks for changes when necessary, reducing unnecessary re-renders.

Commit:
For better readability and more structured code, please move the initialization code into a private method and call it inside ngOnInit(). This will help keep the component's lifecycle hook clean and maintainable.

Commit:
If the function is not used in the HTML template and is intended for internal use only, it's a good practice to add the private access modifier to the function. This ensures that the function is only accessible within the class and isn't exposed for use in the template or elsewhere.

Commit:
Define the return type and access modifier for this method.

Commit:
Define the return type and access modifier for this method.

Commit:
Define the return type and access modifier for this method.

Commit:
Define the return type and access modifier for this method.

Commit:
Define the return type and access modifier for this method.

Commit:
Adding access modifier

Commit:
Adding access modifier

Commit:
Adding access mod

Commit:
Adding access mod

Commit:
Define the return type and access modifier for this method.

Commit:
For better performance, please use the OnPush change detection strategy.

Commit:
For better performance, please use the OnPush change detection strategy.

Commit:
For better performance, please use the OnPush change detection strategy.

Commit:
can use [class.onwalk]="!!walkthroughActive" instead of {{walkthroughActive !== '' ? 'onwalk': ''}} it will be more clean

Commit:
remove empty class

Commit:
Instead of duplicating this list here and in MainScreenComponent, it would be better to define it in a separate file and import it both here and in MainScreenComponent

Commit:
Rename it to IBtnGroupConfig for better naming convention.

Commit:
Rename it to IMainAssigned

Commit:
Missing change detection strategy "onPush"

Commit:
Define the return type and access modifier for this method.

Commit:
remove empty lifecycle method

Commit:
remove bug

Commit:
Fix auto open walthrough stop to work after changeDetection was set to on push on main component

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.html:
```diff
@@ -0,0 +1,34 @@
+    </ng-walkthrough>
+
+    <ng-template #step1Template>
+        <!-- <lib-icon-btn class="closebtn" (click)="close()">
```
Please clean up the commented-out code to keep things tidy and maintainable.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/4a7ba23e509f92c580d6f731e923a870d90e3251

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,368 @@
+    panels:string[] = [];
+    activeScreenId:string = ""; 
+    activeId: string = "";
+    activeArrowId: string = "";
```
Missing access modifiers – component variables should have explicit access modifiers for better readability and maintainability. For example, activeArrowId is only used within the component's TypeScript file, so it should be marked as private. Please review and update similar variables accordingly.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/ef7272fe0acac56420d86b1ed2c0560cb49796f7

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,368 @@
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+      // if(changes['data']){
```
Please clean up the commented-out code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/4a7ba23e509f92c580d6f731e923a870d90e3251

web-fullstack-developer commented on src/app/components/cyrano-walkthrough/cyrano-walkthrough.component.ts:
```diff
@@ -0,0 +1,368 @@
+    }
+
+    private drawArrow(event:WalkthroughComponent, arrowId: string){
+      console.log('drawArrow:',event);
```
Please clean up  console.log statements

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/4a7ba23e509f92c580d6f731e923a870d90e3251

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.html:
```diff
@@ -0,0 +1,55 @@
+                 </lib-icon-btn>
+                <div class="btn-container {{ highlightAll(isActiveScreen(panel)) ? 'highlightall' : ''}}" [class.onwalk]="!!walkthroughActive">
+                    <!-- to be made into a component -->
+                     <div *ngFor="let btngrp of buttonGroup">
```
I meant, is it possible to loop directly on the <app-btn-group> component instead of wrapping it with a <div *ngFor="let btnGrp of buttonGroup">?

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/82e7e99e4e4d322c28c011a276ee7a98a244020f

web-fullstack-developer commented on src/app/components/main-screen/main-screen.component.html:
```diff
@@ -0,0 +1,55 @@
+            </div>
+        </div>
+
+        <!-- <app-tuto-walkthrough></app-tuto-walkthrough> -->
```
Please clean up the commented-out code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/4a7ba23e509f92c580d6f731e923a870d90e3251

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group.component.ts:
```diff
@@ -0,0 +1,128 @@
+
+  private subs = new Subscription();  
+
+  mainAssigned:string = '';
```
Missing access modifiers – component variables should have explicit access modifiers for better readability and maintainability. For example, mainAssigned is only used within the component's TypeScript file, so it should be marked as private. Please review and update similar variables accordingly.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/ef7272fe0acac56420d86b1ed2c0560cb49796f7

web-fullstack-developer commented on src/app/components/shared/btn-group/btn-group.component.ts:
```diff
@@ -0,0 +1,128 @@
+ }
+
+ ngAfterViewInit(){
+  // BtnGroupService.notifyButtonGrpReady(0)
```
clean commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/4a7ba23e509f92c580d6f731e923a870d90e3251

web-fullstack-developer commented on src/app/components/shared/button/button.component.ts:
```diff
@@ -0,0 +1,76 @@
+  @Input() btnSetting:Button;
+  @Input() screenId:string = '';
+
+  buttonActions: { [key:string] : () => void } = {
```
This should be marked as private since it’s only used in the component’s TypeScript file.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/ef7272fe0acac56420d86b1ed2c0560cb49796f7

web-fullstack-developer commented on src/app/components/shared/button/button.component.ts:
```diff
@@ -0,0 +1,76 @@
+      
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
```
Please remove any empty lifecycle methods if they’re not needed—same goes for ngOnInit and ngAfterViewInit, since they’re currently empty, we should clean them up.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/4a7ba23e509f92c580d6f731e923a870d90e3251

web-fullstack-developer commented on src/app/components/shared/button/button.component.ts:
```diff
@@ -0,0 +1,76 @@
+  public buttonClicked(type: string): void{
+    const action = this.buttonActions[type];
+    
+    if (action) {
```
no need t define the "action" variable. it can be if(this.buttonActions[type]) {action()}

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/ef7272fe0acac56420d86b1ed2c0560cb49796f7

web-fullstack-developer commented on src/app/components/shared/button/button.component.ts:
```diff
@@ -0,0 +1,76 @@
+  }
+
+  private openTutorial(): void {
+    console.log("open tutorial...");
```
remove console.log

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/0d2690dd89ee1a1119d9fccce3f6225858395c89

web-fullstack-developer commented on src/app/services/ws.service.ts:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    // ✅ Send a message to the WebSocket server
+    sendMessage(event: string, data: any) {
```
You should define a generic type for the data instead of using any.

nikaqim replied:
Removed not used. Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/43dba3b91306d39f6307a25639f18a532de19217

web-fullstack-developer commented on src/app/services/ws.service.ts:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    // ✅ Listen for messages from the WebSocket server
+    listen(event: string): Observable<any> {
```
same here

nikaqim replied:
Used only for testing. Live json data update. Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/43dba3b91306d39f6307a25639f18a532de19217

web-fullstack-developer commented on src/index.html:
```diff
@@ -0,0 +1,20 @@
+  <link href="https://fonts.googleapis.com/css2?family=Calistoga&display=swap" rel="stylesheet" />
+  <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500&display=swap" rel="stylesheet">
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+  <script src="assets/js/leader-line.min.js"></script>
```
It should be imported in angular.json instead of index.html.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/8eb27ca259573a727c916fb47dd1f5bf4caf1aaf

Commit:
It's recommended to use standalone components instead of NgModule. So, instead of defining the StartScreenModule to declare the StartScreenComponent, you can convert StartScreenComponent into a standalone component.

Commit:
enable auto language update change

nikaqim commented:
All fix are completed. Application is now using standalone component and all feedback are commented with commit url.

Commit:
Clean commented code and console.log

Commit:
revert accidently removed functional code

Commit:
remove console.log

Commit:
missing access mod

Commit:
WS service for testing only

Commit:
move external scripts to angular.json file

Commit:
II meant, is it possible to loop directly on the component instead of wrapping it with a <div *ngFor="let btnGrp of buttonGroup">?

nikaqim commented:
Completed all fixes. Please help to review.

web-fullstack-developer commented on src/app/app.config.ts:
```diff
@@ -0,0 +1,41 @@
+  options: {}
+}
+
+// // Required for AoT
```
There’s still some commented-out code here—mind cleaning it up to keep things tidy?

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/10139fefbb548d6cddec9df0b8413facfe89bbb2

web-fullstack-developer commented on src/app/app.routes.ts:
```diff
@@ -0,0 +1,25 @@
+// import { NgModule } from '@angular/core';
```
same here

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/10139fefbb548d6cddec9df0b8413facfe89bbb2

web-fullstack-developer commented on src/app/services/arrow.service.ts:
```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+
+declare const LeaderLine:any;
```
can't define type instead of 'any'?

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/a78c57840aa3b3eb2fb87fe15f0da038e05e91e3

web-fullstack-developer commented on src/app/services/arrow.service.ts:
```diff
@@ -0,0 +1,76 @@
+
+@Injectable({ providedIn: 'root' })
+export class ArrowService {
+    private lines: Map<string, any> = new Map(); // ✅ Store multiple arrows
```
same here please avoid using 'any'

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/a78c57840aa3b3eb2fb87fe15f0da038e05e91e3

web-fullstack-developer commented on src/app/services/btn.service.ts:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    private loadButtonConfig(): void{
+      this.getButtonConfig().subscribe((data:IBtnGroupConfig) => {
```
You need to define an unsubscribe method for that observable to help avoid memory issues.

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/f61f5ef6a7cab1689cdf543b5e47dcf14759c45c

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,309 @@
+  private walkthroughTextSubject = new BehaviorSubject<CyranoTutorialConfig>({
+  });
+
+  steps:CyranoTutorial[] = [];
```
Access modifiers are missing here

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/5b754ef1eb8c8766c0a3318513762ecde5f66b8e

web-fullstack-developer commented on src/main.ts:
```diff
@@ -0,0 +1,14 @@
+// import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
```
clean commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/10139fefbb548d6cddec9df0b8413facfe89bbb2

web-fullstack-developer commented on src/app/translation.loader.ts:
```diff
@@ -0,0 +1,8 @@
+
+// Required for AoT
+export function HttpLoaderFactory(http: HttpClient): TranslateHttpLoader {
+  // return new TranslateHttpLoader(http, '/home/nikaqim/Workspace/Projects/FIVER/ryannvon-retuto/FVR_RETUTO_PR/src/assets/i18n/', '.json');
```
clean commented code

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/10139fefbb548d6cddec9df0b8413facfe89bbb2

web-fullstack-developer commented on src/app/services/tuto.service.ts:
```diff
@@ -0,0 +1,309 @@
+  }
+
+  public scrollIntoView(elementId:string): void {
+    const parentEl = document.getElementById(elementId);
```
The code can be minimized to be more concise and readable

        document.getElementById(elementId)?.scrollIntoView({
          behavior: 'smooth',
          block: 'center',
          inline: 'center'
        });

nikaqim replied:
Completed -> https://github.com/nikaqim/FVR_RETUTO_PR/pull/1/commits/10139fefbb548d6cddec9df0b8413facfe89bbb2

Commit:
Clean commented code

Commit:
can't define type instead of 'any'?

Commit:
Access modifiers are missing here

Commit:
auto-unsubscribe using take

nikaqim commented:
Completed all fix please review
